osf/models/preprint.py

```diff
@@ -176,6 +177,20 @@ class Preprint(DirtyFieldsMixin, GuidMixin, IdentifierMixin, ReviewableMixin, Ba
     }
     group_format = 'preprint_{self.id}_{group}'
 
+    has_prereg_links = models.NullBooleanField(null=True, blank=True)
+    why_no_prereg = models.TextField(null=True, blank=True)
+    prereg_links = ArrayField(
+        models.URLField(null=True, blank=True),
+        blank=True,
+        null=True
+    )
+
+    prereg_link_explanations = ArrayField(
+        models.CharField(null=True, blank=True, max_length=5000),
```

Comment:
This should be a choice field (same in the API). Choices should be "Pre-registration of study designs, Pre-registration of analysis plans"

Reply:
What do you mean by "same in the API", just that the serializer should have the same?

Reply:
Yup

Reply:
We can't really do a choice field because `ArrayField`s don't allow mixing differently validated `CharFields` within an array, so no `CharFields` in the same array as `UrlFields`. I'm unsure whether this is intentional or just a django bug...

So I kind of went a little nuts here and added a new custom JSON schema field at the model level. So data is stored as JSON like:
`{"urls": [{'url': 'http://osf.io', 'prereg_plan': True, 'prereg_analysis': True}]}`
and is validated against a JSON schema at the model level, not the serializer as we usually do with JSON. Not ideal, but it was either this or a one-to-many IMO.

Is that good?

Reply:
The choice field is a separate field from the URL field. So you have an array of URLs, then you have a separate array of choices. The choice says, "of all of the URLs that you have above, do you have at least one prereg plan and/or one prereg analysis?" So it should just be an array of choices. Does that make sense?